kube-controllers/cmd/kube-controllers/main.go

```diff
@@ -311,6 +300,19 @@ func runHealthChecks(ctx context.Context, s *status.Status, k8sClientset *kubern
 			s.SetReady("KubeAPIServer", true, "")
 		}
 
+		// If we encountered errors, retry again with a longer timeout.
+		if !s.GetReadiness() {
+			timeout = 2 * timeout
+			if timeout > maxTimeout {
+				timeout = maxTimeout
+			}
+			log.Info("Health check is not ready, retrying in 2 seconds with new timeout: %s", timeout)
+			time.Sleep(2 * time.Second)
```

Comment:
Should this be 
```suggestion
			time.Sleep(timeout * time.Second)
```
?

Reply:
Nope, the timeout is for the HTTP calls (used in the context) whereas this is just to prevent tight looping.